### PR TITLE
Display only required properties in sample config

### PR DESCRIPTION
**Pull request by mtuchi**
#### Description
I have update the `sampleConfiguration()` function to only display required properties. 

Ref #471

### CODE REVIEW

**Comment by josephjclark** [comment]
I didn't know this is how we did the sample config. I'm not a fan :grimacing: 

This actually gives us a worse appearance for http right? Because http has no required properties? So the schema will be empty.

I'd much rather have the option to manually define a sample schema somewhere and use that, rather than trying to autogenerate something.

**Comment by mtuchi** [comment]
@josephjclark For adaptor that does not have a required property this is how the sample config will appear 

Screenshot 2024-05-18 at 11 19 18 AM

**Comment by josephjclark** [comment]
Yeah that's my point - it's not very helpful!

"Required" and "example" are two very different things.

Ok, here's a compromise. If the schema has required schema, only show the required properties. Otherwise, show all properties with examples.

I still don't like it, and I'd rather put this time into a good lasting solution. But I'd probably accept it at least!

**Comment by mtuchi** [comment]
I have thought of displaying all properties, but i find it misleading for http because it has both `username, password and token` See screenshot below 👇🏽 . Displaying all of these properties i think it's confusing especially for first time users. 

Should we utilize the `readme.md` For each adaptor and add the content for configuration ? or perhaps create a `configuration.md` file for each adaptor in adaptors monorepo ?
Screenshot 2024-05-20 at 9 27 17 AM

**Comment by josephjclark** [comment]
An optional configuration.md in the adaptor repo would be the ideal. If there isn't one there, we default to showing all (or all required?) properties

**Comment by mtuchi** [comment]
In our adaptor monorepo, currently we have tw `.md` files, `README.md` and `CHANGELOG.md`. We don'f have `configuration.md` file yet. Should i create an issue for adding `configugiration.md` in all adaptors, mean while we display all properties for adaptors that do not have required list ?

**Comment by mtuchi** [comment]
Hiya @josephjclark , I have created an issue for creating the `configuration.md` file in each adaptor, See https://github.com/OpenFn/adaptors/issues/572 and also i have pushed the changes for displaying the full schema if `required` list is `undefined`

File: generate-adaptors/index.js [hunk 1]
```diff
@@ -131,6 +131,16 @@ const sampleConfiguration = json => {
       return conf;
     });
   }
+  if (properties && required) {
+    required.forEach((key, index) => {
+      conf[key] = 'Someting';
```
Comment (josephjclark) [inline generate-adaptors/index.js:136]: Can we remove this line?

File: generate-adaptors/index.js [hunk 2]
```diff
@@ -131,6 +131,16 @@ const sampleConfiguration = json => {
...
+      conf[key] =
```
Comment (josephjclark) [inline generate-adaptors/index.js:137]: Try
```
conf[key] = properties[key].examples?.[0]
```

What happens if there is no example and `conf[key]` is undefined?

Comment (mtuchi) [reply generate-adaptors/index.js]: I think if `conf[key]` is `undefined` that credential schema is not correct and should be investigated.

File: generate-adaptors/index.js (hunk 1 above)
Comment (mtuchi) [reply generate-adaptors/index.js]: ✅ i have replaced that line with what you suggested 
```
conf[key] = properties[key].examples?.[0]
```

**Review by josephjclark — APPROVED** [review]

**Pull request merged**